From 4999e2c32ae345bac9694f0265ddcdc18fcf3f24 Mon Sep 17 00:00:00 2001
From: Neil Armstrong <narmstrong@baylibre.com>
Date: Thu, 6 Sep 2018 18:58:54 +0000
Subject: [PATCH] linux: input: LibInputTouch: Switch to get_seat_slot() to fix
 single touch device input

When a single touch device is connected, the libinput_event_touch_get_slot() call returns -1.
The libinput_event_touch_get_seat_slot() call is designed so return an unique slot number and is valid for single touch devices.

Signed-off-by: Neil Armstrong <narmstrong@baylibre.com>
---
 xbmc/platform/linux/input/LibInputTouch.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/xbmc/platform/linux/input/LibInputTouch.cpp b/xbmc/platform/linux/input/LibInputTouch.cpp
index 7e186df62bcd..fb771cfb1d09 100644
--- a/xbmc/platform/linux/input/LibInputTouch.cpp
+++ b/xbmc/platform/linux/input/LibInputTouch.cpp
@@ -55,7 +55,7 @@ void CLibInputTouch::SetPosition(int slot, CPoint point)
 
 void CLibInputTouch::ProcessTouchDown(libinput_event_touch *e)
 {
-  int slot = libinput_event_touch_get_slot(e);
+  int slot = libinput_event_touch_get_seat_slot(e);
 
   SetPosition(slot, GetPos(e));
   SetEvent(slot, TouchInputDown);
@@ -64,7 +64,7 @@ void CLibInputTouch::ProcessTouchDown(libinput_event_touch *e)
 
 void CLibInputTouch::ProcessTouchMotion(libinput_event_touch *e)
 {
-  int slot = libinput_event_touch_get_slot(e);
+  int slot = libinput_event_touch_get_seat_slot(e);
   uint64_t nanotime = libinput_event_touch_get_time_usec(e) * 1000LL;
 
   SetPosition(slot, GetPos(e));
@@ -78,7 +78,7 @@ void CLibInputTouch::ProcessTouchMotion(libinput_event_touch *e)
 
 void CLibInputTouch::ProcessTouchUp(libinput_event_touch *e)
 {
-  int slot = libinput_event_touch_get_slot(e);
+  int slot = libinput_event_touch_get_seat_slot(e);
 
   SetEvent(slot, TouchInputUp);
   CLog::Log(LOGDEBUG, "CLibInputTouch::%s - touch input up", __FUNCTION__);
@@ -86,7 +86,7 @@ void CLibInputTouch::ProcessTouchUp(libinput_event_touch *e)
 
 void CLibInputTouch::ProcessTouchCancel(libinput_event_touch *e)
 {
-  int slot = libinput_event_touch_get_slot(e);
+  int slot = libinput_event_touch_get_seat_slot(e);
   uint64_t nanotime = libinput_event_touch_get_time_usec(e) * 1000LL;
 
   CLog::Log(LOGDEBUG, "CLibInputTouch::%s - touch input cancel", __FUNCTION__);
